inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,366 @@
+          //We could also change to reference year 0 instead of y-1 if we need it fixed
+          this -> recruitment.phi_zero = 
+            this -> unfished_spawning_biomass[year-1]  / 
+            this -> recruitment.rzero;
+
+          this -> numbers_at_ages[index_ya] = 
+            this -> recruitment -> evaluate(this -> spawning_biomass[year-1]) * this -> recruitment.recruit_deviations[year];
+        }
+
+        //calculate expeted total catch by fleet
+        void CalculateCatch(int year, int age) {
+          for (int fleet_=0; fleet_ < this -> nfleets; fleet_++) {
+            int index_yaf = year*this->nages*this->nfleets + age*this->nfleets + fleet_;
+            int index_yf = year*this->nfleets + fleet_; //index by fleet and years to dimension fold
+            
+            this -> expected_catch[index_yf] += 
+              this -> catch_weight_at_age[index_yaf];
+
+            fleets -> expected_catch[index_yf] += 
+              this -> catch_weight_at_age[index_yaf];
+          }
+        }
+
+        //calculate the index by fleet
+        void CalculateIndex(int index_ya, int year, int age) {
+          for (int fleet_=0; fleet_ < this -> nfleets; fleet_++) {
+            int index_yaf = year*nages*nfleets + age*nfleets + fleet_;
+             //index by fleet and years to dimension fold
+            int index_yf = year*nfleets + fleet_;
+            // I = qN (N is total numbers), I is an index in numbers
+            Type index_;
+            
+            index_ = this -> q * 
+              this -> fleets[fleet_].selectivity->evaluate(age) * 
+              this -> numbers_at_ages[index_ya] *
+              this -> weight_at_age[age];
+              
+            this -> expected_index[index_yf] += index_;
+            fleets -> expected_index[index_yf] += index_;
+          }
+        }
+
+        //don't need separate function for survey - both survey and fishery treated as 'fleet'
+        void CalculateCatchNumbersAA(int index_ya, int year, int age) {
+          for (int fleet_=0; fleet_ < this -> nfleets; fleet_++) {
+            int index_yaf = year*nages*nfleets + age*nfleets + fleet_;
+            int index_yf = year*nfleets + fleet_; //index by fleet and years to dimension fold
+            // make an intermediate value in order to set multiple members (of 
+            // current and fleet objects) to that value.
+            Type catch_; //catch_ is used to avoid using the c++ keyword catch
+            //Baranov Catch Equation
+            catch_ = 
+              (this -> Fmort[index_yf] * 
+              this -> fleets[fleet_].selectivity->evaluate(age) )/
+              this -> mortality_Z[index_ya] * 
+              this -> numbers_at_ages[index_ya] * 
+              (1 - exp(-(this->mortality_Z[index_ya])));
+            this -> catch_numbers_at_age[index_yaf] = catch_;
+            fleets -> catch_numbers_at_age[index_yaf] = catch_;
```

Comment:
this should be fleets -> catch_numbers_at_age[index_yaf] += catch_

Reply:
This is the year, age, and fleet specific expected catch in numbers at age so it doesn't need the +=. The aggregated total catch in weight by year and fleet is being done in CalculateCatch.